shim/src/main.rs

```diff
@@ -338,6 +348,46 @@ fn main() -> Result<()> {
 #[cfg(test)]
 mod test {
     use clap::Parser;
+    use itertools::Itertools;
+
+    struct CommandTestCase {
+        command: &'static str,
+        command_args: Vec<Vec<&'static str>>,
+        global_args: Vec<Vec<&'static str>>,
+        expected_output: Args,
+    }
+
+    impl CommandTestCase {
+        fn test(&self) {
+            let permutations = self.create_all_arg_permutations();
+            for command in permutations {
+                assert_eq!(Args::try_parse_from(command).unwrap(), self.expected_output)
+            }
+        }
+
+        fn create_all_arg_permutations(&self) -> Vec<Vec<&'static str>> {
```

Comment:
Should we also be testing interspersing global args with the command args? If I'm reading this correctly we're only putting global args right after `turbo`. IIRC correctly cobra allows for these args to appear anywhere which is a behavior we want to keep.